### ISSUE TITLE
Multivendor support

### ISSUE THREAD

**Issue by Trudko** [Sylius/Sylius#1346]
Hi, I am implementing multivendor support.

This would require a lot of changes, and because I am not that famialiar with sylius and Symfony any tips would be appreciate.

I was planning to create own bundle which would extends corebundle.

List of changes:
* Create vendor model
Vendor model would represent seller or owner of each shop.
Extend some of models and add relationship with vendor to them.
* Order
* Product
* ShippingMethod
* TaxRate
* PaymentMethod
* Settings

Add new views
* Crud for vendors and own view for admin. Right now admin is owner of the shop and he can see the things which vendor should see.
* ability to deactive some shop

There are certainly other items which need to added.
Is there something which could be a problem while implementing this?

**Comment by a2xchip** [comment]
@Trudko did you implemented this feature?

**Comment by Trudko** [comment]
@a2xchip yeah but don't think but we customized it heavily to our need so I worry its useless for others.

**Comment by a2xchip** [comment]
@Trudko my customer is looking for symfony developer who can create multi vendor market using sylius components. If you are interested in it. You can send me your contacts and I will send it to them.

**Comment by Trudko** [comment]
Thank you for the offer, I really appreciated it. Unfortunately right now I am not available and I won't be for some time.

**Comment by RobertLangore** [comment]
Hi,

what's the status of this feature?
Is anyone working on it?

@pjedrzejewski Will this feature be implemented in a future release?

**Comment by pjedrzejewski** [comment]
@RobertLangore Not in Sylius core, but we have implemented this already and there is a high chance that we will open source it (it is not our code), problem is that it will take time to clean up things for open source release, so I can't promise it will be out anytime soon. :/

**Comment by RobertLangore** [comment]
Ok.
Thanks Pawel.

Please let me know if this feature will be made available (I really hope, because it makes Sylius more complete as an ecommerce platform) as free, open source in the future (even if it is an estimate months ahead). Will it be released after the release of the first stable version of Sylius?

Will it be released as a separate project/repo/bundle not integrated in Sylius? And after such release will it require work to integrate it in Sylius?
Or will it be released as part of Sylius (bundle/feature) fully integrated?

**Comment by comvanilla** [comment]
These could be done In two steps.
1. Dropship
2. Multi Vendor 
Maybe use the approach of Spree, like have an sylius-contrib, where those projects could be hosted.
Dropship solve many users problem, and has an small back log than Multi Vendor, so use it as an step could help on delivery.

**Comment by chathurabandara** [comment]
@pjedrzejewski Do we ave a preview of Multi Store feature?

**Comment by chathurabandara** [comment]
What I am interested is market place approach where multiple sellers can manage their inevntories seperately. Do we have a option for that?

**Comment by gagarine** [comment]
Multi/Single-everything. To be able to cover all scenarios we need multi-store and multi-seller than can be combined with single-store, single-seller. For example, multi-seller can have access to their store (each seller have their own store, like amazon storefront) or to one store but manage their inventory (airbnb). You also want to be able to sell one product on different store (ebay, amazon)... etc..

**Comment by thiriet** [comment]
Hi @pjedrzejewski, any news regarding the multi-store / marketplace features?

**Comment by KirbyFox** [comment]
Hi @pjedrzejewski could you please let me know when the open source Multi Vendor extension will be available?

**Comment by RonaldSc** [comment]
Hi @pjedrzejewski, also looking forward to the multi-store capabilities.

**Comment by ikourfaln** [comment]
Hi guys, please, what's the status of this features, it's really useful
Thanx

**Comment by marcelglaeser** [comment]
Hi at all,
im interesting too! Any news?

**Comment by lovepreetsingh03** [comment]
hii any news regarding the multi-store / marketplace features. and what is the status of this new fetures .

**Comment by gety9** [comment]
Seems like it won't be opensource right ? It's listed as Sylius Plus feature ...

**Comment by marcelglaeser** [comment]
It was first put on hold. :-/

**Comment by ikourfaln** [comment]
Sylius Plus !!
I was waiting for this feature to start a project, but now I noticed that it's not included in Sylius Standard!!

**Comment by marcelglaeser** [comment]
Me too 😓

**Comment by lchrusciel** [comment]
Hello fellows,

To be clear here: Sylius Multivendor Marketplace project has been temporarily suspended. We will come back to this project in the future, but it was planned to be a paid solution. Sylius Plus is also a paid solution, which will come with features like Advanced Multi-Store or Multi-Source Inventory, but without (at least right now) Multivendor functionalities. 

Sylius is still an excellent choice for multivendor marketplaces, but the complexity of these projects make it tough to provide a generic implementation.

Thanks for understanding and I hope Sylius will still provide a good base for your projects.

**Comment by dukusz** [comment]
Hi there!
We've recently sent a newsletter to all Multi-vendor Marketplace mailing list subscribers. If you haven't received it, let me cite it:

> 
### TL;DR
- Sylius Multi-Vendor Marketplace was an experimental, paid product;
- A small group of existing customers will receive patches and support;
- The package will not be available to the wider public in any form, anytime soon;
- For now, we focus our full energy on Sylius for B2C, in the small and medium-sized businesses market.

### What's the backstory?
Shortly after the stable release of the v1.0.0, we were exploring different ideas for the monetization of Sylius project, in order to grow our business and community in perfect harmony. We have used Sylius to build marketplaces ourselves and there have been many requests for such functionality. The concept of Sylius Multi-Vendor Marketplace was born.

### What was the challenge?
Whoever has built a marketplace in their career knows it is the complexity of standard B2C/B2B eCommerce multiplied by at least a factor of 3. We have realized that at the current stage of our company's development, the focus is very important and building two products, for similar but still different markets, is difficult. It requires at least two separate development teams, different communication/content strategies and many more.

### Does it make sense to build a marketplace on top of Sylius?
Definitely! We see lovely marketplaces launched on the top of Open Source version every month (e.g. https://sylius.com/customers/pielers-de/). Our challenge was building a generic enough product and distribute it to customers while doing the same for B2C. Given the unique nature of marketplace projects, Sylius with Symfony is an excellent choice. 

### What's next?
There is still plenty of work to do on Sylius for B2C in order to make it truly the best solution for professional development and we indent to focus on that.

### Sylius Plus
If you are facing a challenge of building a complex marketplace, Sylius Plus: https://sylius.com/plus/ might still be a good choice - i.e. things like Returns Management or Multi-source Inventory could be useful. With our knowhow provided within Project Success Assistance, we are able to help with custom marketplace projects. ;)

**Comment by odiseoteam** [comment]
Hello to all,

In case anyone is still interested, we have a sylius plugin that allows you to create a marketplace store, you can see more information here: https://odiseo.io/marketplace

I hope it works. Regards!

**Comment by bitbager** [comment]
Don't know if the thread is still live. There is also an open-source project that was recently published, which uses Sylius as a core extending it with MVM features. Here is the link: https://open-marketplace.io/
